From d02c6bc9a5a7da44d1dd59e6a37ba822d7d02e6d Mon Sep 17 00:00:00 2001
From: Srinu Jella <sjella@codeaurora.org>
Date: Mon, 27 Aug 2018 17:26:25 +0530
Subject: [PATCH 4/4] Bluetooth: Ensure to release wake lock during turn off

- Timers running with less than 3 seconds will acquire
  wake lock internally.

- This patch ensures to release wake lock as part of
  turn off procedure if any module has not stopped
  timers properly.

CRs-Fixed: 2310057
Change-Id: I384dfc47d2e57371066efd370bbdca36e5910eb4
---
 osi/src/wakelock.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/osi/src/wakelock.cc b/osi/src/wakelock.cc
index 46c462caa..389c68159 100644
--- a/osi/src/wakelock.cc
+++ b/osi/src/wakelock.cc
@@ -213,6 +213,10 @@ static void wakelock_initialize_native(void) {
 }
 
 void wakelock_cleanup(void) {
+  if (wakelock_stats.is_acquired) {
+    LOG_ERROR(LOG_TAG, "%s releasing wake lock as part of cleanup", __func__);
+    wakelock_release();
+  }
   wake_lock_path.clear();
   wake_unlock_path.clear();
   initialized = PTHREAD_ONCE_INIT;
-- 
2.17.1

